text/0121-config-service.md

```diff
@@ -0,0 +1,165 @@
+
+Our remote configuration protocol will support this call:
+
+```
+service DynamicConfig {
+  rpc GetConfig (ConfigRequest) returns (ConfigResponse);
+}
+```
+
+A request to the config service will look like this:
+
+```
+message ConfigRequest{
+
+  // Required. The resource for which configuration should be returned.
+  opentelemetry.proto.resource.v1.Resource resource = 1;
+
+  // Optional. The value of ConfigResponse.fingerprint for the last configuration
+  // a resource received that was successfully applied.
+  bytes last_known_fingerprint = 2;
+}
+```
+
+While the response will look like this:
+
+```
+message ConfigResponse {
+
+  // Optional. The fingerprint associated with this ConfigResponse. Each change
+  // in configs yields a different fingerprint.
+  bytes fingerprint = 1;
+
+  // Dynamic configs specific to metrics
+  message MetricConfig {
+
+
+    // A Schedule is used to apply a particular scheduling configuration to
+    // a metric. If a metric name matches a schedule's patterns, then the metric
+    // adopts the configuration specified by the schedule.
+
+    message Schedule {
+
+      // A light-weight pattern that can match 1 or more
+      // metrics, for which this schedule will apply. The string is used to
+      // match against metric names. It should not exceed 100k characters.
+      message Pattern {
+        oneof match {
+          string equals = 1;       // matches the metric name exactly
+          string starts_with = 2;  // prefix-matches the metric name
+        }
+      }
+
+      // Metrics with names that match at least one rule in the inclusion_patterns are
+      // targeted by this schedule. Metrics that match at least one rule from the
+      // exclusion_patterns are not targeted for this schedule, even if they match an
+      // inclusion pattern.
+
+      // For this iteration, since we only want one Schedule that applies to all metrics,
+      // we will not check the inclusion_patterns and exclusion_patterns.
+      repeated Pattern inclusion_patterns = 1;
+      repeated Pattern exclusion_patterns = 2;
+
+      // CollectionPeriod describes the sampling period for each metric. All
+      // larger units are divisible by all smaller ones.
+      enum CollectionPeriod {
+        NONE = 0;  // For non-periodic data (client sends points whenever)
+        SEC_1 = 1;
+        SEC_5 = 5;
+        SEC_10 = 10;
+        SEC_30 = 30;
+        MIN_1 = 60;
+        MIN_5 = 300;
+        MIN_10 = 600;
+        MIN_30 = 1800;
+        HR_1 = 3600;
+        HR_2 = 7200;
+        HR_4 = 14400;
+        HR_12 = 43200;
+        DAY_1 = 86400;
+        DAY_7 = 604800;
+      }
+      CollectionPeriod period = 3;
```

Comment:
Is this enum needed, if the number is in seconds? Given that the values are in seconds, it implies that unlisted values (e.g., 2 days == 172800) may be used. If unlisted values can be used, do we need these constants?

How is NONE meant to be used?

Reply:
The enum is used to enforce the property that all larger periods must be multiples of the smaller ones. This might be useful in that we can batch the export of metrics with less frequent CollectionPeriods with those with more frequent CollectionPeriods. 

Just using an int definitely definitely provides more flexibility. Let me know what you think!

So I deleted the comment left with `NONE`. The intention of `NONE` is just to say "do not export". Might be useful for the user to have an option they can toggle to stop exports for the user. `NONE` can be gotten rid of if it seems unnecessary.

Reply:
Since the server supplies this configuration to the SDK, it can have whichever validation on multiples that it wants, without having that reflected in the protocol schema. So +1 to just have a duration field. The field name should reflect that it is in seconds.

Reply:
Done!